scripts/analysis.py

```diff
@@ -904,65 +913,96 @@ def commod_per_inst(cur, commodity, timestep=10000):
 
     Returns
     -------
-    inst_output_dict: dictionary
+    institution_output: dictionary
         key = institution
         value = timeseries list of outflux of commodity
     """
 
-    institutions = get_inst(cur)
-    inst_output_dict = collections.OrderedDict()
-    for inst in institutions:
+    institutes = institutions(cur)
+    institution_output = collections.OrderedDict()
+    for inst in institutes:
         inst_id = inst[1]
         inst_name = inst[0]
         facilities = cur.execute('SELECT agentid FROM agententry '
                                  'WHERE parentid = ' + str(inst_id)).fetchall()
-        facilities_list = []
+        facilities_collected = []
         for fac in facilities:
-            facilities_list.append(fac[0])
-        query = exec_string(facilities_list, 'senderid', 'sum(quantity)')
+            facilities_collected.append(fac[0])
+        query = exec_string(facilities_collected, 'senderid', 'sum(quantity)')
         query += ' AND commodity = "' + commodity + \
             '" and time < ' + str(timestep)
-        inst_output_dict[inst_name] = cur.execute(query).fetchone()[0]
+        institution_output[inst_name] = cur.execute(query).fetchone()[0]
 
-    return inst_output_dict
+    return institution_output
 
 
-def get_waste_dict(isotope_list, mass_list, time_list, duration):
+def wastes(isotopes, mass_timeseries, duration):
     """Given an isotope, mass and time list, creates a dictionary
        With key as isotope and time series of the isotope mass.
 
-    Parameters
-    ----------
-    isotope_list: list
+    Inputs:
+    isotopes: list
         list with all the isotopes from resources table
-    mass_list: list
-        list with all the mass values from resources table
-    time_list: list
-        list with all the time values from resources table
-    duration: int
+    mass_timeseries: list
+        a list of lists.  each outer list corresponds to a different isotope
+        and contains tuples in the form (time,mass) for the isotope transaction.
+    duration: integer
         simulation duration
 
-    Returns
-    -------
-    waste_dict: dictionary
+    Outputs:
+    waste: dictionary
         dictionary with "key=isotope, and
         value=mass timeseries of each unique isotope"
     """
-    waste_dict = collections.OrderedDict()
-    isotope_set = set(isotope_list)
-    for iso in isotope_set:
-        mass = 0
-        time_mass = []
-        # at each timestep,
-        for i in range(0, duration):
-            # for each element in database,
-            for x, y in enumerate(isotope_list):
-                if i == time_list[x] and y == iso:
-                    mass += mass_list[x]
-            time_mass.append(mass)
-        waste_dict[iso] = time_mass
 
-    return waste_dict
+    keys = []
+    for key in isotopes:
+        keys.append(key)
+
+    waste = {}
```

Comment:
This is all alone, surrounded by blank space, and isn't used anywhere nearby. Move it down to where it's used.